azurerm/resource_arm_postgresql_server.go

```diff
@@ -0,0 +1,306 @@
+								50,
+								100,
+							}),
+						},
+
+						"tier": {
+							Type:     schema.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(postgresql.Basic),
+							}, true),
+							DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+						},
+					},
+				},
+			},
+
+			"administrator_login": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"administrator_login_password": {
+				Type:      schema.TypeString,
+				Required:  true,
+				Sensitive: true,
+			},
+
+			"version": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(postgresql.NineFullStopFive),
+					string(postgresql.NineFullStopSix),
+				}, true),
+				DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+				ForceNew:         true,
+			},
+
+			"storage_mb": {
+				Type:     schema.TypeInt,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validateIntInSlice([]int{
+					51200,
+					102400,
+				}),
+			},
+
+			"ssl_enforcement": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(postgresql.SslEnforcementEnumDisabled),
+					string(postgresql.SslEnforcementEnumEnabled),
+				}, true),
+				DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+			},
+
+			"fqdn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceArmPostgreSQLServerCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).postgresqlServersClient
+
+	log.Printf("[INFO] preparing arguments for AzureRM PostgreSQL Server creation.")
+
+	name := d.Get("name").(string)
+	location := d.Get("location").(string)
+	resGroup := d.Get("resource_group_name").(string)
+
+	adminLogin := d.Get("administrator_login").(string)
+	adminLoginPassword := d.Get("administrator_login_password").(string)
+	sslEnforcement := d.Get("ssl_enforcement").(string)
+	version := d.Get("version").(string)
+	storageMB := d.Get("storage_mb").(int)
+
+	tags := d.Get("tags").(map[string]interface{})
+
+	sku := expandAzureRmPostgreSQLServerSku(d, storageMB)
+
+	properties := postgresql.ServerForCreate{
+		Location: &location,
+		Sku:      sku,
+		Properties: &postgresql.ServerPropertiesForDefaultCreate{
+			Version:                    postgresql.ServerVersion(version),
+			StorageMB:                  azure.Int64(int64(storageMB)),
+			SslEnforcement:             postgresql.SslEnforcementEnum(sslEnforcement),
+			AdministratorLogin:         azure.String(adminLogin),
+			AdministratorLoginPassword: azure.String(adminLoginPassword),
+			CreateMode:                 postgresql.CreateModeDefault,
+		},
+		Tags: expandTags(tags),
+	}
+
+	_, error := client.CreateOrUpdate(resGroup, name, properties, make(chan struct{}))
+	err := <-error
+	if err != nil {
+		return err
+	}
+
+	read, err := client.Get(resGroup, name)
```

Comment:
As questioned elsewhere - is it necessary to use real ID instead of saving `name` as ID? There's risk of dangling resource in case of a failure.

Reply:
Yep, given this is the ID Azure exposes to users (and is used throughout the provider) - so exposing something different at this point would cause more issues than it solves